###################################################################################
#####        params for ep_qrcode_loc
#####           yuanxun@ep-ep.com
#####        updateTime:  2024.10.20
###################################################################################

ep_qrcode_loc:
#----------------------------------------------------------------------------------
  # 运行模式1:采集二维码坐标（长时间停在二维码上方）; 
  # 运行模式2:采集二维码编号（快速扫过）；
  # 运行模式3:正常运行；
  # 运行模式4:只扫码,不递推,用于测试；
  # 运行模式5:采集地码角度补偿数据
  # 运行模式6:计算地码补偿角度
  operating_mode: 3
#----------------------------------------------------------------------------------
  # 日志级别 "FATAL" "ERROR" "WARN" "INFO" "DEBUG"
  logLevel: "INFO"
#----------------------------------------------------------------------------------
  # 调试程序过程中用到参数
  show_original_msg: false # 记录和发布相机扫到二维码的原始信息
  is_pub_tf: false # 后续删除相关参数和代码
  avliable_yaw: 177.0 # 可用方向角
  is_debug: true
  cfg_dir: "/home/zl/work/ep-qrcode-loc/src/ep_qrcode_loc/config/"
#----------------------------------------------------------------------------------
  # 地码方向角补偿相关
  err_ratio_offline: 1.0 # 地码方向角补偿率
#----------------------------------------------------------------------------------
  # 扫码跳变时，递推值与观测值加权平均的权重
  rec_p1: 0.4 #递推值权重，观测值权重自动计算
#----------------------------------------------------------------------------------
  # 跳变检测
  yaw_jump_UL: 2.0    #角度跳变限制(度)
  x_jump_UL: 0.05 #车身纵向跳变限制(m)
  y_jump_UL: 0.05 #车身横向跳变限制(m)
#----------------------------------------------------------------------------------
  # 轮速参数
  wheel_diameter: 0.343 # 轮子直径
  wheel_reduction_ratio: 20.0 # 主动轮电机转速比
  wheel_base_dis: 1.524 # 主动轮与后轮之间的轴距
  wheel_angular_offset: 0.0  # 主动轮零位偏移，单位：度
#----------------------------------------------------------------------------------
  # 低速界限,运行模式3中,高于该速度时每次只取一帧二维码数据
  low_speed_UL: 0.1
#----------------------------------------------------------------------------------
  # 最大递推距离,超多该距离则将输出帧标记为不可用
  maxEstimationDis: 1.1 
#----------------------------------------------------------------------------------
  # 相机网络所用端口,与相机内设置的一致
  port: 1024        
#----------------------------------------------------------------------------------
  # 库位设置中的相关尺寸
  detect_site_dis: 1.9      # 识别点距离
  aux_site_dis: 1.4         # 辅助点距离
  forkaction_site_dis: 0.93 # 货叉动作点距离
  site_site_dis: 1.36       # 库位前后距离